src/extractor/guidance/turn_analysis.cpp

```diff
@@ -1659,87 +1663,84 @@ std::vector<TurnCandidate> TurnAnalysis::getTurnCandidates(const NodeID from_nod
     const NodeID only_restriction_to_node =
         restriction_map.CheckForEmanatingIsOnlyTurn(from_node, turn_node);
     const bool is_barrier_node = barrier_nodes.find(turn_node) != barrier_nodes.end();
+    bool has_uturn_edge = false;
 
     for (const EdgeID onto_edge : node_based_graph.GetAdjacentEdgeRange(turn_node))
     {
-        bool turn_is_valid = true;
+        // reverse edges are never valid turns because the resulting turn would look like this:
+        // from_node --via_edge--> turn_node <--onto_edge-- to_node
         if (node_based_graph.GetEdgeData(onto_edge).reversed)
         {
-            turn_is_valid = false;
+            continue;
```

Comment:
This breaks turn handling, since invalid turns are part of the turn situation but are not reported anymore